web/libs/frontend-test/src/helpers/LSF/AudioView.ts

```diff
@@ -334,6 +348,325 @@ class AudioViewHelper extends withMedia(
+              hasContent = true;
+              break;
+            }
+          }
+
+          if (hasContent) {
+            cy.log("🎨 Canvas has content!");
+            return cy.wrap(null);
+          }
+          cy.log("🔍 Canvas empty, waiting for content...");
+          cy.wait(200); // Increased wait for CI environments
+          return checkForContent();
+        });
+      };
+
+      cy.log("🏁 Waiting for canvas content...");
+      return checkForContent();
+    }
+
+    /**
+     * Waits for audio and video elements to be synchronized
+     * @param tolerance tolerance for time/rate differences
+     * @param timeout maximum time to wait
+     * @param checkVideo whether to check video sync (default: true)
+     */
+    waitForMediaSync(tolerance = 0.01, timeout = 5000, checkVideo = true) {
+      const startTime = Date.now();
+
+      const checkSync = (): Cypress.Chainable => {
+        if (Date.now() - startTime > timeout) {
+          cy.log(`⏰ Media sync timeout after ${timeout}ms`);
+          return cy.wrap(null);
+        }
+
+        if (checkVideo) {
+          // Audio+Video mode: Must sync both elements
+          return cy.get("audio").then(([audio]) => {
+            return cy.get("video").then(([video]) => {
+              const timeDiff = Math.abs(audio.currentTime - video.currentTime);
+              const rateDiff = Math.abs(audio.playbackRate - video.playbackRate);
+
+              if (timeDiff <= tolerance && rateDiff <= tolerance) {
+                cy.log(`🎯 Media sync achieved! Time diff: ${timeDiff.toFixed(3)}s, Rate diff: ${rateDiff.toFixed(3)}`);
+                return cy.wrap(null);
+              }
+              cy.log(`🔄 Media syncing... Time diff: ${timeDiff.toFixed(3)}s, Rate diff: ${rateDiff.toFixed(3)}`);
+              cy.wait(50);
+              return checkSync();
+            });
+          });
+        }
+        // Audio-only mode - no sync needed
+        cy.log("🎯 Audio-only mode - sync achieved!");
+        return cy.wrap(null);
+      };
+
+      cy.log(`🏁 Waiting for ${checkVideo ? "audio/video" : "audio-only"} synchronization...`);
+      return checkSync();
+    }
+
+    /**
+     * Waits for audio/video to be in a specific play state
+     * @param shouldBePlaying expected play state
+     * @param timeout maximum time to wait
+     * @param checkVideo whether to check video state (default: true)
+     */
+    waitForPlayState(shouldBePlaying: boolean, timeout = 8000, checkVideo = true) {
+      cy.log(
+        `🎵 Waiting for ${checkVideo ? "audio/video" : "audio"} to ${shouldBePlaying ? "start playing" : "be paused"}...`,
+      );
+
+      return cy
+        .get("audio", { timeout })
+        .should(([audio]) => {
+          expect(audio.paused).to.equal(!shouldBePlaying);
+        })
+        .then(() => {
+          if (checkVideo) {
+            return cy.get("video").should(([video]) => {
```

Comment:
For checks that include more than one tag, I would suggest to use approach with [`SyncGroup`](https://github.com/HumanSignal/label-studio/blob/develop/web/libs/frontend-test/src/helpers/utils/media/SyncGroup.ts) or similar.